pulp_rpm/app/serializers/repository.py

```diff
@@ -97,11 +99,25 @@ class RpmPublicationSerializer(PublicationSerializer):
         choices=CHECKSUM_CHOICES,
         default=CHECKSUM_TYPES.SHA256,
     )
+    gpgcheck = GPGCheckOptionField(
```

Comment:
I am not sure I see a lot of value defining GPGCheckOptionField.
How about using IntegerField `serializers.IntegerField(max_value=1, min_value=0)`

The error is explicit enough as well:
```
(pulp) [vagrant@pulp2-nightly-pulp3-source-centos7 _scripts]$ http POST "$BASE_ADDR"/pulp/api/v3/publications/rpm/rpm/     repository=/pulp/api/v3/repositories/rpm/rpm/00b82101-5416-42c5-a48a-a3b7e927f46a/ repo_gpgcheck=10
HTTP/1.1 400 Bad Request
Allow: GET, POST, HEAD, OPTIONS
Connection: close
Content-Length: 67
Content-Type: application/json
Date: Thu, 15 Oct 2020 09:50:59 GMT
Server: gunicorn/20.0.4
Vary: Accept, Cookie
X-Frame-Options: SAMEORIGIN

{
    "repo_gpgcheck": [
        "Ensure this value is less than or equal to 1."
    ]
}

(pulp) [vagrant@pulp2-nightly-pulp3-source-centos7 _scripts]$ http POST "$BASE_ADDR"/pulp/api/v3/publications/rpm/rpm/     repository=/pulp/api/v3/repositories/rpm/rpm/00b82101-5416-42c5-a48a-a3b7e927f46a/ repo_gpgcheck=lala
HTTP/1.1 400 Bad Request
Allow: GET, POST, HEAD, OPTIONS
Connection: close
Content-Length: 50
Content-Type: application/json
Date: Thu, 15 Oct 2020 09:51:03 GMT
Server: gunicorn/20.0.4
Vary: Accept, Cookie
X-Frame-Options: SAMEORIGIN

{
    "repo_gpgcheck": [
        "A valid integer is required."
    ]
}
```
wdyt?

Reply:
I created this field because I thought that it would be better to have
```
    gpgcheck = GPGCheckOptionField(
        help_text=_(
            "An option specifying whether a client should perform "
            "a GPG signature check on packages."
        ),
    )
    repo_gpgcheck = GPGCheckOptionField(
        help_text=_(
            "An option specifying whether a client should perform "
            "a GPG signature check on the repodata."
        ),
```
instead of
```
    gpgcheck = serializers.IntegerField(
        max_value=1,
        min_value=0,
        default=0,
        required=False,
        help_text=_(
            "An option specifying whether a client should perform "
            "a GPG signature check on packages."
        ),
    )
    repo_gpgcheck = serializers.IntegerField(
        max_value=1,
        min_value=0,
        default=0,
        required=False,
        help_text=_(
            "An option specifying whether a client should perform "
            "a GPG signature check on the repodata."
        ),
    )
```

The only benefit of this field is to have the attributes, which these GPG check options have in common, in one place.

Reply:
Still, I am removing the field.